client/reporter.go

```diff
@@ -0,0 +1,20 @@
+package client
+
+import (
+	"context"
+	"io"
+)
+
+type Reporter interface {
```

Comment:
There seems to be a logical separation between these method sets so perhaps they should be distinct interfaces to create granularity? They can always be composed into a super interface providing this complete set.

Reply:
This is mostly based on the former `Client` interface from `/packet/client.go`. I'm not 100% sure all the places these calls are made, but you might be right. For this PR, I didn't want to change the packet API calls as much as I could, so I might put that into a future PR